osquery/tables/system/darwin/smc_keys.cpp

```diff
@@ -379,28 +382,99 @@ inline uint32_t strtoul(const char *str, size_t size, size_t base) {
   return total;
 }
 
-float getConvertedValue(const std::string &smcType, const std::string &smcVal) {
-  // Convert hex string to decimal.
+inline uint64_t strtoull(const char* str, size_t size, size_t base) {
+  uint64_t total = 0;
+  for (size_t i = 0; i < size; i++) {
+    total += static_cast<uint64_t>(static_cast<unsigned char>(str[i]))
+             << (size - 1 - i) * 8;
+  }
+  return total;
+}
```

Comment:
Companion to `strtoul`, but for 64 bit ints.

Reply:
Other parts of our code use `std::strtoull`. Should this?

Reply:
I don't think so.  The `std:strtoull` function does what you'd probably expect from the name: it takes a numeric string and interprets it as a number, e.g. the string `"12345"` becomes the unsigned long `12345`.  The `strtoul` function defined in smc_keys.cpp does something else entirely: it treats a string as a set of bytes representing an integer with big-endian encoding.  It's functionally the same as byte-swapping a string and `memcpy`ing the bytes into a new `uint32_t`.  I don't know why it was called `strtoul`, but for consistency I named the 64-bit version the same thing.  I'd be happy to rename them both, but without knowing the history I prefer to stay low-touch.

Reply:
I agreee consistency with the local function makes sense, though the divergence from `std::` seems unfortunate.